xdsl/transforms/convert_pdl_to_pdl_interp/conversion.py

```diff
@@ -788,3 +788,40 @@ def _nodes_match(self, node: MatcherNode, predicate: OrderedPredicate) -> bool:
         return (
             node.position == predicate.position and node.question == predicate.question
         )
+
+    def _insert_exit_node(self, root: MatcherNode) -> MatcherNode:
+        """Insert exit node at end of failure paths"""
+        curr = root
+        while curr.failure_node:
```

Comment:
We try to specify this explicitly in case objects decided to implement `__bool__` in the future and cause unexpected behaviour
```suggestion
        while curr.failure_node is not None:
```